schematic/store/synapse.py

```diff
@@ -659,105 +670,36 @@ def getFilesInStorageDataset(
         Raises:
             ValueError: Dataset ID not found.
         """
-        # select all files within a given storage dataset folder (top level folder in
-        # a Synapse storage project or folder marked with contentType = 'dataset')
-        walked_path = synapseutils.walk(
-            self.syn, datasetId, includeTypes=["folder", "file"]
-        )
+        file_list = []
 
-        current_entity_location = self.synapse_entity_tracker.get(
-            synapse_id=datasetId, syn=self.syn, download_file=False
-        )
+        # Identify all folders nested under the dataset folder
+        folders = synapseutils.walk(self.syn, datasetId, includeTypes=["folder"])
```

Comment:
>  Would building a temporary view based on the dataset folder be helpful here instead of this?

I think it could be used here as well. I think it might be useful to explore building a temporary file view to use for `SynapseStorage` objects in cases where `dataset_scope` is used instead of utilizing the entire view as it exists on synapse. We'd then have all the information we need for the operation locally and wouldn't have to do repeated queries, and in this case we'd be able to avoid walking the directory and querying the view again to get the files by just using the temporary view.

In the past we've only scoped by project; now that `dataset_scope` is in production as an option we might want to make the switch to leverage the ability to set the scope of the temporary view since that isn't an option when we just query the existing complete view and we have to do these workarounds for folders and files. But this all seemed like a separate issue from changing the source of the paths. I'd love to know y'all's thoughts.

> One other thought if you had the fileview, is using the path and do something like this ..
>
> select * from .... where path like 'myproject/myfolder/path/%'
>
> That should get you all the files under a folder

I had considered a solution like that too, I thought gathering the files based on direct parent would help avoid anything weird that could happen with the paths but I think both are viable.

Reply:
Before merging, we do need to discuss this.  I think there are projects with over 100 nested folders that this current function would break.

Reply:
> In the past we've only scoped by project; now that dataset_scope is in production as an option we might want to make the switch to leverage the ability to set the scope of the temporary view since that isn't an option when we just query the existing complete view and we have to do these workarounds for folders and files.

From Toms initial investigation the performance is quickly flips over to being more efficient for the temporary view. What I am not a fan of is that this means we need to:

1. Build a view (in some project)
2. Wait for an index of that view to occur to update the synapse project
3. Query it
4. tear down the view



Querying an existing file view that contains the dataset you're looking at seems like the easier thing to do here, and executing the `select * from .... where path like 'myproject/myfolder/path/%'` query. This is closer to how SQL is expected to be used. Us hacking around needing to create temporary views I feel like is a "brittle" solution, with a lot of moving parts where if any individual piece fails, fails the entire transaction.